core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -1471,11 +1471,13 @@ default Optional<CatalogSchemaTableName> redirectTable(ConnectorSession session,
         return Optional.empty();
     }
 
+    @Deprecated
```

Comment:
maybe add a comment that it's now used only in test code and is not needed for any engine logic cc @findepi

Reply:
Unused methods should be removed from here.

For test-only functionality, this should be handled purely on the test side

Reply:
i don't see a reason to keep unused method in ConnectorMetadata.
if someone wants to support two versions at a time, they can still compile their `ConnectorMetadata` impl (just without `@Override` annotation)

anyway, don't feel strongly about one version of not, just please do follow up. you should be able to create commit/PR without waiting for anything.

Reply:
@findepi I think we have one-version-backward compatibility policy

Reply:
@gaurav8297 can you put up a PR that will remove these deprecated methods? to be merged after 422 goes out

Reply:
looks like this is part of https://github.com/trinodb/trino/pull/18561